packages/runtime/container-runtime/src/opLifecycle/opDecompressor.ts

```diff
@@ -44,12 +45,18 @@ export class OpDecompressor {
             const asObj = JSON.parse(intoString);
```

Comment:
A comment for existing code on line 31: Why do we need both message.metadata?.compressed & message.compression? It feels like the latter is sufficient, no? Having two properties results in having to deal with weird combos (for example, I do not understand what it means to have message.metadata?.compressed === true, but no message.compression - from the code it sounds like we simply treat it as binary payload encoded with base64. Not sure if we need this format, but we can track it with message.compression === "base64", if needed)

Also, does PUSH has deployed code to rountrip message.compression? What happens if we run it against AFR or any other service that does not rountrip compression property? I.e. do we have defenses to present error state to OCE in easy to consume way?

Reply:
> A comment for existing code on line 31: Why do we need both message.metadata?.compressed & message.compression? It feels like the latter is sufficient, no? 

Not sure. Will check with Justus

Reply:
Opened ADO:2897 to deal with this in a separate PR.